ctru-rs/examples/ir-user.rs

```diff
@@ -0,0 +1,251 @@
+            PACKET_BUFFER_SIZE,
+            PACKET_COUNT,
+            PACKET_BUFFER_SIZE,
+            PACKET_COUNT,
+        )
+        .expect("Couldn't initialize ir:USER service");
+        println!("ir:USER service initialized");
+
+        // Get event handles
+        let connection_status_event = ir_user
+            .get_connection_status_event()
+            .expect("Couldn't get ir:USER connection status event");
+        let receive_packet_event = ir_user
+            .get_recv_event()
+            .expect("Couldn't get ir:USER recv event");
+
+        Self {
+            top_console,
+            bottom_console,
+            ir_user,
+            connection_status_event,
+            receive_packet_event,
+        }
+    }
+
+    fn print_status_info(&self) {
+        self.top_console.select();
+        self.top_console.clear();
+        println!("{:#x?}", self.ir_user.get_status_info());
+        self.bottom_console.select();
+    }
+
+    fn connect_to_cpp(&self, hid: &Hid) -> ConnectionResult {
+        // Connection loop
+        loop {
+            hid.scan_input();
+            if hid.keys_held().contains(KeyPad::KEY_START) {
+                return ConnectionResult::Canceled;
+            }
+
+            // Start the connection process
+            self.ir_user
+                .require_connection(IrDeviceId::CirclePadPro)
+                .expect("Couldn't initialize circle pad pro connection");
+
+            // Wait for the connection to establish
+            if let Err(e) = self
+                .connection_status_event
+                .wait_for_event(Duration::from_millis(100))
+            {
+                if !e.is_timeout() {
+                    panic!("Couldn't initialize circle pad pro connection: {e}");
+                }
+            }
+
+            self.print_status_info();
+            if self.ir_user.get_status_info().connection_status == 2 {
+                println!("Connected!");
+                break;
+            }
+
+            // If not connected (ex. timeout), disconnect so we can retry
+            self.ir_user
+                .disconnect()
+                .expect("Failed to disconnect circle pad pro connection");
+
+            // Wait for the disconnect to go through
+            if let Err(e) = self
+                .connection_status_event
+                .wait_for_event(Duration::from_millis(100))
+            {
+                if !e.is_timeout() {
+                    panic!("Couldn't initialize circle pad pro connection: {e}");
+                }
+            }
+        }
+
+        // Sending first packet retry loop
+        loop {
+            hid.scan_input();
+            if hid.keys_held().contains(KeyPad::KEY_START) {
+                return ConnectionResult::Canceled;
+            }
+
+            // Send a request for input to the CPP
+            if let Err(e) = self
+                .ir_user
+                .request_input_polling(CPP_CONNECTION_POLLING_PERIOD_MS)
```

Comment:
it might be nice to have these APIs take a `std::time::Duration` instead, and convert (or fail) to ms internally?

Reply:
I'll leave it as-is because we have a very limited range of period values allowed (one unsigned byte). Doing it this way ensures that only valid inputs are given.